docs/advanced/misc.rst

```diff
@@ -304,3 +304,31 @@ the default settings are restored to prevent unwanted side effects.
 
 .. [#f4] http://www.sphinx-doc.org
 .. [#f5] http://github.com/pybind/python_example
+
+.. _avoid-cpp-types-in-docstrings:
+
+Avoid C++ types in docstrings
+=============================
+
+Most of docstrings are generated at the time of the declaration, e.g. when ``.def(...)`` is called.
```

Comment:
Are there docstrings that aren't?

Reply:
`Enum.__doc__` is read only property, so it's content evaluated dynamically. I thought there was something else...

Reply:
Looks like enum is the only exception and probably it's irrelevant in this context. Let's remove "Most of " part ?

Reply:
Right, OK, yes! Just curious.

What about "Almost all docstrings", then, to emphasize this is practically always?

Reply:
(ok, py::module is a another one, but again, it's irrelevant enough to not mention)